esphome/config_validation.py

```diff
@@ -1185,6 +1183,16 @@ def ipaddress(value):
     return address
 
 
+def ipaddress_multi_broadcast(value):
+    try:
+        address = ip_address(value)
+    except ValueError as exc:
+        raise Invalid(f"{value} is not a valid IP address") from exc
+    if not (address.is_multicast or (address == IPv4Address("255.255.255.255"))):
+        raise Invalid(f"{value} is not a multicast address nor local broadcast address")
+    return address
```

Comment:
```suggestion
    address = ip_address(value)
    if address.is_multicast or address == IPv4Address("255.255.255.255"):
        return address
    raise Invalid(f"{value} is neither a multicast address nor the local broadcast address")
```
`ip_address` already raises the appropriate exception.